### PR TITLE
changes types to please gcc6.

### PR DESCRIPTION
**Pull request by schmidh**
Mostly type changes to make gcc6 a happy camper. CreEPS.cpp had includes in namespaces which gcc6 does not like any more. For system commands in systemSpeakExample I casted the result to (void) which is idiomatic c/c++ to subpress unused return value warnings. But gcc6 ignores it! At least one can see that everything is meant to be right when you compile the code. There seems to be a discussion in the gcc community how to handle this warning. Here is the official documentation https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html saying that it should work. And here is an old ticket which has been updated lately: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425

### CODE REVIEW

**Comment by schmidh** [comment]
Actually, I think return values of *getNumInputChannels* and *getNumFrames* should be changed from size_t to unsigned int. Please let me know.

File: addons/ofxNetwork/src/ofxTCPClient.cpp [hunk 1]
```diff
@@ -298,7 +298,7 @@ int ofxTCPClient::receiveRawMsg(char * receiveBuffer, int numBytes){
 	int posDelimiter = findDelimiter(tmpBuffReceive.getData(),tmpBuffReceive.size(),messageDelimiter);
 	if(posDelimiter>0){
 		memcpy(receiveBuffer,tmpBuffReceive.getData(),posDelimiter);
-		if(tmpBuffReceive.size() > posDelimiter + (int)messageDelimiter.size()){
+		if(tmpBuffReceive.size() > (unsigned int) posDelimiter + messageDelimiter.size()){
```
Comment (bakercp) [inline addons/ofxNetwork/src/ofxTCPClient.cpp]: size_t here?

**Comment by bakercp** [comment]
Besides my small comment, this looks good to me.  Casting the return value to `void` feels odd, but it isn't "wrong".

File: addons/ofxNetwork/src/ofxTCPClient.cpp (hunk 1 above)
Comment (schmidh) [reply addons/ofxNetwork/src/ofxTCPClient.cpp]: You mean the *posDelimiter*? Looks strange to change a delimiter position to size_t.

**Comment by schmidh** [comment]
Casting a return value to void is idiomatic C/C++. Please see the link to the documentation.

See also e.g. https://stackoverflow.com/a/689716/365425:
> ... covers the motivation for this, to explicitly show other "developers" that you know this function returns but you're explicitly ignoring it.

**Comment by arturoc** [comment]
@schmidh even if it's common practice it might look strange to beginners who are the main target for examples so perhaps actually using the return value as the warning suggests is a better solution?

something like:

```
if(system(...) != 0){
    ofLogError() << "error while running ...";
}
```

we also have ofSystem which runs the command and returns the output instead so we could just use that

**Comment by schmidh** [comment]
Oh yes, I like that a lot!

**Comment by schmidh** [comment]
Using `ofSystem(cmd.c_str())` now.

**Comment by arturoc** [comment]
thanks

**Pull request merged**